app/models/group.rb

```diff
@@ -47,6 +46,16 @@ def add_leader_to_members
     members << leader unless members.include? leader
   end
 
+  def validate_suite_size_inclusion
+    if draw.present?
+      return if draw.suite_sizes.include? size
+      errors.add :size, 'must be a suite size included in the draw'
+    else
+      return if SuiteSizesQuery.call.include? size
+      errors.add :size, 'must be a valid suite size'
+    end
+  end
```

Comment:
Yea... excellent catch. I'll add a comment to this validation method (maybe with a TODO header just to keep us on our toes). Let me know what you think!